prog/any.go

```diff
@@ -126,7 +126,7 @@ func (target *Target) isAnyRes(name string) bool {
 
 func (target *Target) CallContainsAny(c *Call) (res bool) {
 	ForeachArg(c, func(arg Arg, ctx *ArgCtx) {
-		if target.isAnyPtr(arg.Type()) {
+		if target.isAnyPtr(arg.Type()) || res {
```

Comment:
What does this additional condition do?

Reply:
It's not strictly necessary for the PR, I just noticed it while exploring the code.

`ctx.Stop` only acts on the currently explored subtree, but once `res` is `true` there's no longer any reason to go into any subtrees at all.

Reply:
Should foreachArgImpl then check ctx.Stop before callback as well?

Otherwise we will need to fix all iteration callbacks to do this. But this really looks like the issue in iteration functions, rather than something to bother each callback with.

Reply:
I had an impression that it's actually the intended semantics, e.g. here we only want to stop going deeper into the tree:

https://github.com/google/syzkaller/blob/e8e3023880fc4f8be85c1d1c689fd1fc9a77d60e/prog/resources.go#L50-L54

And here

https://github.com/google/syzkaller/blob/e8e3023880fc4f8be85c1d1c689fd1fc9a77d60e/prog/mutation.go#L224-L228

Reply:
OK, then, you probably know this code better than me now :)